src/ff/galois_field.rs

```diff
@@ -71,7 +71,8 @@ macro_rules! bit_array_impl {
                 }
 
                 fn truncate_from<T: Into<u128>>(v: T) -> Self {
-                    let v = &v.into().to_le_bytes()[..<Self as Serializable>::Size::to_usize()];
+                    const MASK: u128 = u128::MAX >> (u128::BITS - <$name>::BITS);
+                    let v = &(v.into() & MASK).to_le_bytes()[..<Self as Serializable>::Size::to_usize()];
                     Self(<$store>::new(v.try_into().unwrap()))
                 }
```

Comment:
I'm not sure what you're suggesting... I'm sorry - I'm missing something.

Reply:
I believe what Martin is saying is that `v.into()` allocates the intermediate 16 bytes when we are converting values from `v: T` where T is Into<u128> to e.g., GF(2^8). I'm not sure how we can bind T to numeric types though. Maybe use `[num_traits](https://docs.rs/num-traits/latest/num_traits/trait.Num.html)`?

Reply:
I'll elaborate...

You don't need specific implementations: this one is entirely generic and could be part of the definition of `Field`.  If the assumption is that truncation is able to drop high order bits because the caller has ensured that the value fits (I think that is the assumption) then this only requires that the implementation supports `try_from()`.

Of course, `TryFrom` for GF uses this method, a dependency order that would need to be inverted for that to work.  But the upshot would be that Field contains an implementation that is generic and specific implementations could write optimized versions.  For `Gf32Bit` that implementation could take the `u128,` truncate to `u32` and put those values in.

That leads me to ask: why are we still using `BitArray` here?  Won't this work for integers?

Reply:
Initially, `truncate_from()` was equivalent to `Field::from()`; an infallible conversion used to draw a random value from PRSS. Now that we have prime fields and Galois fields both sub types of `Field`, I changed `PrimeField::from()` to `PrimeField::truncate_from()` for consistency. Don't they need specific implementations? PrimeField takes modulo prime, and GaloisField just ignores high order bits.

I think `BitArray` is useful here if we ever need to create GF(2^n) if `n` is not a multiple of 8. It lets us store bits in u8's, and accessing bits is (they claim) faster.

Reply:
I'm inclined to believe the claim if they are inclined to make it.  No need to change anything here for now.